AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/AWSS3StoragePlugin.swift

```diff
@@ -5,21 +5,36 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-import Amplify
+@_spi(InternalAmplifyConfiguration) import Amplify
 import Foundation
 import AWSPluginsCore
+import InternalAmplifyCredentials
 
 /// The AWSS3StoragePlugin which conforms to the Amplify plugin protocols and implements the Storage
 /// Plugin APIs for AWS S3.
 ///
 /// - Tag: AWSS3StoragePlugin
 final public class AWSS3StoragePlugin: StorageCategoryPlugin {
 
-    /// An instance of the S3 storage service.
-    var storageService: AWSS3StorageServiceBehavior!
+    /// The default S3 storage service.
+    var defaultStorageService: AWSS3StorageServiceBehavior! {
```

Comment:
will these cause an exception if the service is force_un

Reply:
Yes, but that's the case with the majority of variables in all of our plugins. 
We're basically giving them the outlets treatment: they will be `nil` at initialization, but will be set before the plugins are actually used (on their `configure` methods).
We could make them all optional but then we'd have to unwrap them every single time we want to use them. If a plugin is not configured (i.e. these variables are `nil`), we throw an exception anyway before reaching any of these variables.